### PR TITLE
basic sanity checking for parameter fixing

### PR DESCRIPTION
**Pull request by FFroehlich**
really easy to mix up `x_fixed_vals` and `x_fixed_indices`.

### CODE REVIEW

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ pypesto/problem.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5775146&bid=18901466)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ pypesto/problem.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5775146&bid=18901466)

File: pypesto/problem.py [hunk 1]
```diff
@@ -383,3 +392,54 @@ def print_parameter_summary(self) -> None:
...
+    }
+}
+
+
+def _type_conversion_with_check(index: int,
+                                value: Union[SupportsFloat, SupportsInt],
+                                valuename: str,
+                                convtype: str) -> Union[float, int]:
+    """
+    Converts values to the requested type, raises and appropriate error if
```
Comment (yannikschaelte) [inline pypesto/problem.py:414]:
```suggestion
    Converts values to the requested type, raises an appropriate error if
```

**Comment by FFroehlich** [comment]
> makes sense. is it possible to still have the, e.g. fixed values to be an int list? I think it would be good to still allow passing ints there which are then converted to float. maybe add a test?

Yes, should still be possible. Will add test.

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ pypesto/problem.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5775146&bid=18901466)

**Comment by FFroehlich** [comment]
turned out that tests were quite helpful in squashing a couple more bugs.

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- test/test_problem.py  4
         

Complexity decreasing per file
==============================
+ pypesto/problem.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5775146&bid=18901466)

**Pull request merged**